modules/dns_common/dns_common.rb

```diff
@@ -17,9 +19,28 @@ def initialize(server = nil, ttl = nil)
     end
 
     def resolver
+      logger.warn("Proxy::Dns::Record#resolver has been deprecated and will be removed, use get_dns_resource(s)")
```

Comment:
No, I am not breaking the compatibility, I am deprecating the functionality with a plan to remove (break) it. That's a difference.

One possible solution to avoid modification of plugins is to provide a decorator object that would keep the API. If you insist, I can do that, however I think that it's hacky solution. No other plugin uses that and I am willing to patch route53 until the method is deprecated.

Reply:
IMHO this method still makes sense. I'd be ok with deprecating it and making it private if there's a good reason solution to route53, but duplicating the creation of a resolver object is something I'd like to avoid.

Reply:
But what do you recommend then? I believe that defining a better contract (two methods) is so much better than just a method which creates new instance.

Reply:
Use all 3 methods. The new methods should be recommended. I'd also copy the caching of the instance that route53 does so it becomes a method that lazy loads the resolver instance.